### ISSUE TITLE
[BUG] 2.18 requires both configs for PKCS setup plugins.security.ssl.transport.keystore_password and ...keystore_keypassword

### ISSUE THREAD

**Issue by AntonEliatra** [opensearch-project/security#4961]
**What is the bug?**
OS 2.18 requires both configs for PKCS setup plugins.security.ssl.transport.keystore_password and ...keystore_keypassword

**How can one reproduce the bug?**
Remove one of the 2 lines (keystore_password or keystore_keypassword) and try to start the cluster:

`plugins.security.ssl.transport.enforce_hostname_verification: false
plugins.security.ssl.transport.keystore_filepath: opensearch-node.jks
plugins.security.ssl.transport.keystore_alias: opensearch-node
**plugins.security.ssl.transport.keystore_password: <password>**
**plugins.security.ssl.transport.keystore_keypassword: <password>** #This is required option in 2.18.0
plugins.security.ssl.transport.truststore_filepath: opensearch-truststore.jks
plugins.security.ssl.transport.truststore_alias: root-ca
plugins.security.ssl.transport.truststore_password: truststorepassword

plugins.security.ssl.http.enabled: true
plugins.security.ssl.http.keystore_filepath: opensearch-node.jks
plugins.security.ssl.http.keystore_alias: opensearch-node
**plugins.security.ssl.http.keystore_password: <password>**
**plugins.security.ssl.http.keystore_keypassword: <password>**  #This is required option in 2.18.0
plugins.security.ssl.http.truststore_filepath: opensearch-truststore.jks
plugins.security.ssl.http.truststore_alias: root-ca
plugins.security.ssl.http.truststore_password: truststorepassword`

**What is the expected behavior?**
Only one of these configuration is required

**What is your host/environment?**
 - Version 2.18.0

**Comment by cwperks** [comment]
[Triage] @willyborankin Can you take a look at this issue?

**Comment by cwperks** [comment]
@willyborankin I haven't dug into this in great depth, but is this because of the default value here? https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/ssl/config/SslCertificatesLoader.java#L77

**Comment by willyborankin** [comment]
@cwperks, you’re right—this is the root cause. Refactoring unfortunately breaks backward compatibility.

**Comment by cwperks** [comment]
Addressed in https://github.com/opensearch-project/security/pull/4999
